scripts/prepare_release.rb

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env ruby
+
+require 'octokit'
+require 'fileutils'
+require 'erb'
+require 'colorize'
+require 'optparse'
+require 'open3'
+
+def prompt_for_new_version
+  puts "What's the new version number?"
+  version = gets.chomp
+end
+
+def validate_version_number(version_number)
+  part_names = ['major', 'minor', 'patch']
+  parts = version_number.split('.')
+
+  unless parts.length() == 3
+    abort("Invalid version number. It should consists of a major, minor, and patch number.")
+  end
+
+  parts.each_with_index do | part, index |
+    if part.start_with?('0') && part.length() > 1
+      part_name = part_names[index]
+      abort("Invalid version number: #{part_name} number can\'t begin with 0.")
+    end
+  end
+end
+
+def ensure_clean_repo()
+  repo_dir = File.basename(Dir.getwd)
+  if repo_dir != "stripe-android"
+    abort("You must run this script from 'stripe-android'.")
+  end
+
+  stdout, stderr, status = Open3.capture3("git status --porcelain")
+
+  if !stdout.empty?
+    abort("You must have a clean working directory to continue.")
+  end
+end
+
+def update_changelog(version)
+  done = false
+  final_lines = []
+
+  File.foreach('CHANGELOG.md') do |line|
+    is_line = line.include? "XX.XX.XX"
+    final_lines.append(line)
+
+    date = Time.now.strftime("%Y-%m-%d")
+
+    if is_line
+        final_lines.append("\n")
+        final_lines.append("## #{version} - #{date}\n")
+    end
+  end
+
+  File.write('CHANGELOG.md', final_lines.join(""))
+end
+
+def github_login
+  token = `fetch-password -q bindings/gh-tokens/$USER`
+  if $?.exitstatus != 0
+    puts "Couldn't fetch GitHub token. Follow the Android SDK Deploy Guide (https://go/android-sdk-deploy) to set up a token. \a".red
+    exit(1)
+  end
+  client = Octokit::Client.new(access_token: token)
+  abort('Invalid GitHub token. Follow the wiki instructions for setting up a GitHub token.') unless client.login
+  client
+end
+
+def open_url(url)
+  `open '#{url}'`
+end
+
+def pull_current_master
+  system("git checkout master")
+  system("git pull")
```

Comment:
If a command fails, does execution of the entire script stop, or do we just continue executing?  If we continue to just execute, i think we need to check the return value of each command and halt appropriately and/or clean up if needed.

Reply:
Fixed now with `execute_or_fail`.